### PR TITLE
camera: only request flight info when system has autopilot

### PR DESCRIPTION
**Pull request by dlech**
Currently, the camera plugin assumes that the system was an autopilot and always requests flight information. However it may be the case that the system is not an autopilot, e.g. a stand-alone camera.

This adds a condition to check that the system is an autopilot before sending the request.

### CODE REVIEW

**Review by julianoes — COMMENTED** [review]
Thanks for the pull request.

I can see how this change would be beneficial, however, I think there is one catch to it. If a camera gets discovered before the autopilot it will then never try to get that information, and this then means that the state is not updated later.

Actually, now that I checked what this is for, I'm ok with this change. This is really only for backwards compatibility with Yuneec drones. Support for these will probably have to be dropped soon anyway.

**Review by julianoes — DISMISSED** [review]

**Review by JonasVautherin — APPROVED** [review]
Thanks!

**Comment by JonasVautherin** [comment]
Oh, style is failing. Can you run the script? `./tools/fix_style.sh .` should do it.

**Comment by dlech** [comment]
style is fixed

File: src/plugins/camera/camera_impl.cpp [hunk 1]
```diff
@@ -201,7 +209,10 @@ void CameraImpl::manual_disable()
 {
     invalidate_params();
     _parent->remove_call_every(_camera_information_call_every_cookie);
-    _parent->remove_call_every(_flight_information_call_every_cookie);
+
+    if (_parent->has_autopilot()) {
+        _parent->remove_call_every(_flight_information_call_every_cookie);
```
[the remark this answers is not in the document]
Comment (JonasVautherin) [reply src/plugins/camera/camera_impl.cpp:215]: I'm sorry I don't see the change about the cookie :thinking:

File: src/plugins/camera/camera_impl.cpp [hunk 2]
```diff
@@ -201,7 +210,10 @@ void CameraImpl::manual_disable()
 {
     invalidate_params();
     _parent->remove_call_every(_camera_information_call_every_cookie);
-    _parent->remove_call_every(_flight_information_call_every_cookie);
+
+    if (_flight_information_call_every_cookie) {
```
Comment (dlech) [inline src/plugins/camera/camera_impl.cpp:214]: > I'm sorry I don't see the change about the cookie 

here it is

File: src/plugins/camera/camera_impl.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlech) [reply src/plugins/camera/camera_impl.cpp:215]: this comment thread still show the original change

**Pull request merged**